account/rethinkdb/store.go

```diff
@@ -0,0 +1,279 @@
+package rethinkdb
+
+import (
+	"fmt"
+
+	rdb "github.com/dancannon/gorethink"
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/go-errors/errors"
+	"github.com/ory-am/common/pkg"
+	"github.com/ory-am/hydra/account"
+	"github.com/ory-am/hydra/hash"
+	"github.com/pborman/uuid"
+)
+
+const accountsTable = "hydra_account"
+
+type Store struct {
+	hasher  hash.Hasher
+	session *rdb.Session
+}
+
+func New(h hash.Hasher, session *rdb.Session) *Store {
+	return &Store{h, session}
+}
+
+func (s *Store) CreateTables() error {
+	exists, err := s.tableExists(accountsTable)
+	if err == nil && !exists {
+		_, err := rdb.TableCreate(accountsTable).RunWrite(s.session)
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+	return nil
+}
+
+// TableExists : check if table(s) exists in database
+func (s *Store) tableExists(table string) (bool, error) {
+
+	res, err := rdb.TableList().Run(s.session)
+	if err != nil {
+		return false, err
```

Comment:
handling errors in go is painful because we're missing important stack information for debugging. Througout hydra I have used [go-errors](https://github.com/go-errors/errors) to wrap "native" error results. It would be good if you could do so in the rethink implementation as well.